### ISSUE TITLE
Custom Redo keybinding doesn't work.

### ISSUE THREAD

**Issue by mbomb007** [HeidiSQL/HeidiSQL#943]
## Steps to reproduce this issue

1. Go to Tools -> Preferences, Shortcuts, SQL Editing
2. Find "Redo". There are two of them in the list. (What's the difference?)
3. Set the secondary keybinding for one or both to Ctrl+Y and Apply changes.
4. Set the keybinding for "Delete Line" to something else, like "Ctrl+Shift+D".
5. Click OK
6. (optionally restart HeidiSQL)
7. Modify something in the editor
8. Ctrl+Z to undo
9. Ctrl+Y to redo (doesn't work)

## Current behavior

Ctrl+Y will never work for Redo-ing an action that was undone.

## Expected behavior

It should work.

## Possible solution

Fix the Shortcuts Keybindings. They don't work well and it's hard to know what keybindings are used where if there is a conflict.

## Environment

* **HeidiSQL version:** 10.3.0.5918 (64-bit)
* **Operating system:** Windows 10

**Comment by RaabenF** [comment]
yes, that's the most annoying thing about heidi to me, i would like to change.
wonder if this is due to synedit?

does anywone know why there is still Synedit v1 used?

**Comment by ansgarbecker** [comment]
It looks like it works somehow, when I apply a different shortcut to the "Delete line" command, not just remove it. Afterwards I can apply `Ctrl+Y` to `Redo`, and it works like expected here, regardless of the log messages. Am I blind or what is going on here?

**Comment by mbomb007** [comment]
> It looks like it works somehow, when I apply a different shortcut to the "Delete line" command, not just remove it. Afterwards I can apply `Ctrl+Y` to `Redo`, and it works like expected here, regardless of the log messages. Am I blind or what is going on here?

It still doesn't work for me. Not only that, but Ctrl+Shift+Z doesn't redo either.

Windows 11 Build 26100
HeidiSQL version 12.15.0.7171 (64 bit)

**Comment by jabcreations** [comment]
There is a framework being used that clearly manipulates keyboard shortcuts, Delphi? I'm guessing they also release a lot of updates; I think it would be fair to summize that Ansgar is keeping the library up-to-date and therefore some of the tested environments work and others do not. I'd recommend including the framework version somewhere accessible especially since only so many users will change shortcuts and then out of those only so many will hit the bug and then out of those looking and/or filing a bug report.
